### PR TITLE
add #let A = B #in C syntax

### PR DESCRIPTION
**Pull request by dwightguth**
Turns out to be pretty simple.

### CODE REVIEW

**Review by radumereuta — CHANGES_REQUESTED** [review]
This could use a test.
If you can piggyback on an existing one that would be even better.
Other than that, the changes to the code look fine.

**Comment by dwightguth** [comment]
Whoops. Good catch. I wrote a test and apparently forgot to `git add` it

**Review by radumereuta — APPROVED** [review]
I will let @ehildenb add the automerge if he is happy with the functionality.

File: k-distribution/tests/regression-new/let-test/test.k [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module TEST
+  imports INT
+
+  rule 0 => #let FOO = 1 #in FOO
```
Comment (ehildenb) [inline k-distribution/tests/regression-new/let-test/test.k:6]: Let's add a few more rules, specifically one with 1 nested binding, one with 2. If you are happy with the output they give, then feel free to add automerge @dwightguth

File: kernel/src/main/java/org/kframework/compile/RewriteAwareVisitor.java [hunk 2]
```diff
@@ -69,6 +69,18 @@ public void apply(KApply k) {
                 isRHS = wasRHS;
                 isLHS = wasLHS;
                 apply(k.items().get(1));
+            } else if (k.klabel().name().equals("#fun3")) {
```
Comment (ehildenb) [inline kernel/src/main/java/org/kframework/compile/RewriteAwareVisitor.java]: Should this be `#fun3` or `#let`? I guess the next case was handling `#fun3` before, and now this case is handling `#fun3` and the next case is handling `#let`.

Comment (ehildenb) [reply kernel/src/main/java/org/kframework/compile/RewriteAwareVisitor.java]: Ah, actually, the cases are identical. Why don't we just do `.equals('#fun3') or .equals('#let')` and have one case?

**Comment by nishantjr** [comment]
@dwightguth Could we add some documentation please.

**Comment by nishantjr** [comment]
@dwightguth @ehildenb Are we also thinking about syntax for allowing let bindings that works across cells as well?
Looking at this I don't think we can do:

```
rule <foo> I => Foo </foo>
      <bar> _ => Foo </bar>
      where Foo :=K foo(I)
```

**Pull request merged**